utils/windowfunction/windowfunctiontype.cpp

```diff
@@ -596,6 +556,15 @@ void WindowFunctionType::implicit2T(uint64_t i, T& t, int s)
             break;
         }
     }
+
+    s -= fRow.getScale(i); // we scale only the difference of scales
```

Comment:
For DOUBLE/UDOUBLE/FLOAT/UFLOAT/LONGDOUBLE, we didn't have `s -= fRow.getScale(i);` in the individual case statements above. Is this still OK for these datatypes?

Reply:
The old code for double:
            if (s == 0)
                t = (T) fRow.getDoubleField(i);
            else
                t = (T) (fRow.getDoubleField(i) * IDB_pow[s]); // s is scale, [0, 18]
The only difference here is that scale is assumed to be positive because, for some reason, we're not taking the diff.
That being said, the only time s is non zero is when getting the default for LEAD|LAG (if not a constant). Currently, the MDB parser doesn't parse the default parameter for LEAD|LAG, so there's no way to test it. This is sleeping code -- unreachable -- until MDB  implements the full standard for Window Functions. Nonetheless, I changed a line in LEAD|LAG that I think looks wrong.